sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecuteHolder.scala

```diff
@@ -243,6 +243,9 @@ private[connect] class ExecuteHolder(
    *   true if it was not interrupted before, false if it was already interrupted.
    */
   def interrupt(): Boolean = {
+    if (eventsManager.status == ExecuteStatus.Pending) {
+      return false
```

Comment:
1. According to the function description, `false` is already occupied for the status where it was already interrupted.

2. For the code change, according to the state transition, can we change the status to `ExecuteStatus.Canceled` status directly from the current `ExecuteStatus.Pending` because it's not started yet? In this case, we can return `true`.

https://github.com/apache/spark/blob/79a0ca72daca8be70950a8fe8b986a1d7ff8a6f7/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecuteEventsManager.scala#L38-L47

Reply:
> For the code change, according to the state transition, can we change the status to ExecuteStatus.Canceled status directly from the current ExecuteStatus.Pending because it's not started yet? In this case, we can return true.

Actually, that was my first idea to solve this issue. But as I mentioned in the description, I found that didn't work because transitioning from `Pending` to `Canceled` causes another issue.

Reply:
Got it. Thank you. As long as the code and description are consistent, I'm okay for both. (1) Updating the description by changing the meaning of `false` and (2) changing the return types.

Reply:
Thank you for your suggestion. I'll simply update the description.